src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -218,49 +219,53 @@ async def test_TC_BRBINFO_4_1(self):
+        await self._send_keep_active_command(stay_active_duration_ms, dynamic_endpoint_id)
 
         logging.info("Waiting for ActiveChanged from DUT...")
-        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+        timeout_s = idle_mode_duration_s + max(active_mode_duration_ms, stay_active_duration_ms)/1000
+        promised_active_duration_ms = await self._wait_for_active_changed_event(timeout_s)
 
-        asserts.assert_greater_equal(promised_active_duration, stay_active_duration, "PromisedActiveDuration < StayActiveDuration")
+        asserts.assert_greater_equal(promised_active_duration_ms, stay_active_duration_ms,
+                                     "PromisedActiveDuration < StayActiveDuration")
 
         self.step("2")
 
-        stay_active_duration = 1500
-        logging.info(f"Sending KeepActiveCommand({stay_active_duration}ms)")
-        self._send_keep_active_command(stay_active_duration)
+        stay_active_duration_ms = 1500
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration_ms}ms)")
+        await self._send_keep_active_command(stay_active_duration_ms, dynamic_endpoint_id)
 
         logging.info("Waiting for ActiveChanged from DUT...")
-        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+        timeout_s = idle_mode_duration_s + max(active_mode_duration_ms, stay_active_duration_ms)/1000
+        promised_active_duration_ms = await self._wait_for_active_changed_event(timeout_s)
 
         # wait for active time duration
-        time.sleep(max(stay_active_duration/1000, promised_active_duration))
+        sleep_time_s = max(stay_active_duration_ms, promised_active_duration_ms)/1000
+        time.sleep(sleep_time_s)
         # ICD now should be in idle mode
 
         # sends 3x keep active commands
-        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
-        self._send_keep_active_command(stay_active_duration, dynamic_endpoint_id)
-        time.sleep(100)
-        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
-        self._send_keep_active_command(stay_active_duration, dynamic_endpoint_id)
-        time.sleep(100)
-        logging.info(f"Sending KeepActiveCommand({stay_active_duration})")
-        self._send_keep_active_command(stay_active_duration, dynamic_endpoint_id)
-        time.sleep(100)
+        logging.info(f"Step3 Sending first KeepActiveCommand({stay_active_duration_ms})")
+        await self._send_keep_active_command(stay_active_duration_ms, dynamic_endpoint_id)
+        time.sleep(0.1)
+        logging.info(f"Step3 Sending second KeepActiveCommand({stay_active_duration_ms})")
+        await self._send_keep_active_command(stay_active_duration_ms, dynamic_endpoint_id)
+        time.sleep(0.1)
```

Comment:
I'm not sure what the hig sleep was before, so I have no idea if this change makes sense. Why was the sleep so long before? Is the idea to send three commands before the event can be generated? If so, this feels racy.

Reply:
That is because it is racy. I have a plan to address this (aka remove these sleeps completely) in a follow up PR in test plan spec and implementation

Reply:
Here is the first test plan change:
https://github.com/CHIP-Specifications/chip-test-plans/pull/4512

Another small tweak will be made in a follow up if https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10236 improvement lands